test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,539 @@
+        redeemRequest.gasLimit,
+        redeemRequest.blockNumber,
+        redeemRequest.hashLock,
+        redeemRequest.storageProof,
+      );
+      await Utils.expectRevert(
+        eip20Gateway.confirmRedeemIntent(
+          redeemRequest.redeemer,
+          nonce.addn(1),
+          redeemRequest.beneficiary,
+          redeemRequest.amount,
+          redeemRequest.gasPrice,
+          redeemRequest.gasLimit,
+          redeemRequest.blockNumber,
+          redeemRequest.hashLock,
+          redeemRequest.storageProof,
+        ),
+        "Previous process is not completed.",
+      );
+    }
+  );
+
+  it('should emit `RedeemIntentConfirmed` event.', async function () {
+
+    let tx = await eip20Gateway.confirmRedeemIntent(
+      redeemRequest.redeemer,
+      redeemRequest.nonce,
+      redeemRequest.beneficiary,
+      redeemRequest.amount,
+      redeemRequest.gasPrice,
+      redeemRequest.gasLimit,
+      redeemRequest.blockNumber,
+      redeemRequest.hashLock,
+      redeemRequest.storageProof,
+    );
+
+    let event = EventDecoder.getEvents(tx, eip20Gateway);
+
+    assert.strictEqual(
+      tx.receipt.status,
+      true,
+      'Transaction should success.',
+    );
+
+    assert.isDefined(
+      event.RedeemIntentConfirmed,
+      'Event RedeemIntentConfirmed must be emitted.',
+    );
+
+    let eventData = event.RedeemIntentConfirmed;
+
+    let params = TestDataJSON.gateway.confirm_redeem_intent.params;
+    let messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+
+    assert.strictEqual(
+      eventData._messageHash,
+      messageHash,
+      `Message hash from event must be equal to ${messageHash}.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemer,
+      params.redeemer,
+      `Redeemer address from event must be equal to ${params.redeemer}.`,
+    );
+
+    let nonce = new BN(params.nonce, 16);
+    assert.strictEqual(
+      nonce.eq(eventData._redeemerNonce),
+      true,
+      `Redeemer nonce from event must be equal to ${nonce.toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      eventData._beneficiary,
+      params.beneficiary,
+      `Beneficiary address from event must be equal to ${params.beneficiary}.`,
+    );
+
+    let amount = new BN(params.amount, 16);
+    assert.strictEqual(
+      amount.eq(eventData._amount),
+      true,
+      `Amount from event must be equal to ${amount.toString(10)}.`,
+    );
+
+    let blockHeight = new BN(params.blockNumber, 16);
+    assert.strictEqual(
+      blockHeight.eq(eventData._blockHeight),
+      true,
+      `Block height from event must be equal to ${blockHeight.toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      eventData._hashLock,
+      params.hashLock,
+      `Hash lock from event must be equal to ${params.hashLock}.`,
+    );
+
+  });
+
+  it('should confirm new redeem intent if status of previous ' +
+    'confirmed redeem intent is revoked', async function () {
```

Comment:
```suggestion
    'redeem message is revoked', async function () {
```

Similar change across the test.